### PR TITLE
CMR-6812 Code Coverage Reports During CI Builds

### PR DESCRIPTION
**Pull request by jaybarra**
Introduced kaocha and cloverage to generate code coverage reports during automated builds. 
Kaocha is a test library that uses the existing clojure.test engine to run tests but provides plugins for junit output and cloverage reports.

Some tests were touched to fix missing `is` statements and tagged with metadata to denote if they require an external database connection or in-memory-db.

There will be additional artifacts produced during the builds to capture each modules code coverage report.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -263,6 +263,22 @@ Once these are in place, you will be able to run the following:
...
+
+Unit Tests:
+```
+lein modules utest
+```
+
+Integration Tests against a running CMR system:
+```
+# If running CMR with the in-memory database (default)
+lein modules itest --skip-meta :oracle
```
Comment (mschmele) [inline README.md:276]: Do you have these commands swapped?

File: es-spatial-plugin/test/cmr/elasticsearch/plugins/spatial/plugin_test.clj [hunk 2]
```diff
@@ -0,0 +1,5 @@
+(ns cmr.elasticsearch.plugins.spatial.plugin-test
+  (:require
+   [clojure.test :refer :all]))
+
+(deftest ^:kaocha/pending spatial-plugin-test)
```
Comment (mschmele) [inline es-spatial-plugin/test/cmr/elasticsearch/plugins/spatial/plugin_test.clj:5]: Maybe add a comment about why this file is here. It's hard to infer based on the contents of the file

File: oracle-lib/test/cmr/oracle/sql_utils_test.clj [hunk 3]
```diff
@@ -0,0 +1,5 @@
+(ns cmr.oracle.sql-utils-test
+  (:require
+   [clojure.test :refer :all]))
+
+(deftest ^:kaocha/pending utils-placeholder-test)
```
Comment (mschmele) [inline oracle-lib/test/cmr/oracle/sql_utils_test.clj:5]: Similar to the comment on the spatial test - maybe have a comment about why kaocha requires placeholder tests

File: redis-utils-lib/tests.edn [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+ :color? #profile {:default true
+                   :ci false}
+
+ :reporter #profile {:default kaocha.report.progress/report
+                     :ci kaocha.report/documentation}
+
+ :kaocha.plugin.junit-xml/target-file "target/junit.xml"
+
+ :kaocha.plugin.profiling/count 3
```
Comment (mschmele) [inline redis-utils-lib/tests.edn:23]: I've seen this in all of the `tests.edn` files and it's kind of a magic number. Maybe a comment about why 3 is the right number here would be useful

File: system-int-test/test/cmr/system_int_test/search/collection_temporal_search_test.clj [hunk 5]
```diff
@@ -26,70 +26,131 @@
 ;; values of the min and max granule of the collection or the collection's temporal values if the
 ;; collection has no granules
 (deftest search-by-temporal-limit-to-granules
-  (let [coll1 (d/ingest-umm-spec-collection "PROV1" (data-umm-c/collection 1 {:EntryTitle "coll1"
```
Comment (mschmele) [inline system-int-test/test/cmr/system_int_test/search/collection_temporal_search_test.clj:29]: This looks much nicer now :)

**Review by mschmele — APPROVED** [review]
A few minor comments mostly just looking for added documentation

File: README.md (hunk 1 above)
Comment (jaybarra) [reply README.md:276]: skip-meta will ignore anything with the tags provided. The command will skip tests requiring a live connection.

Comment (jceaser) [reply README.md:276]: where is the full list of tags?

Comment (jaybarra) [reply README.md:276]: Those are the only two tags available that are specific to CMR. Kaocha has :pending and :skip available but those aren't important to a user running the tests.

**Review by jceaser — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jceaser) [reply README.md:276]: I think the tests you can run should be listed, new developers will not know what is possible.

**Pull request merged**